client/src/app/queries/assessments.ts

```diff
@@ -210,3 +216,45 @@ const removeSectionOrderFromQuestions = (
     })),
   };
 };
+
+export const useFetchAssessmentsWithArchetypeApplications = () => {
+  const { assessments, isFetching: assessmentsLoading } = useFetchAssessments();
+
+  const archetypeQueries = useQueries({
+    queries:
+      assessments?.map((assessment) => ({
+        queryKey: ["archetype", assessment.archetype?.id],
+        queryFn: () =>
+          assessment.archetype?.id
+            ? getArchetypeById(assessment.archetype.id)
+            : undefined,
+        enabled: !!assessment.archetype?.id,
+      })) || [],
```

Comment:
Probably a bit cleaner if you do a filter across the assessments to only return the ones with an archetype and then generate the queries for those.  It'll make the query itself simpler (no undefiend or enabled handling then).  Eg. 
```js
  assessments
    .map((a) => a?.archetype?.id)
    .filter(Boolean)
    .map((archetypeId) => ({ ..... })
```

Maybe even an id dedupe after the filter in there in case multple archetype ids are listed together so only 1 query gets generated per archetype id?